### ISSUE TITLE
Cabal shouldn't try to build split-objs when ghc reports it does not support it

### ISSUE THREAD

**Issue by bos** [haskell/cabal#414]
(Imported from [Trac #421](http://hackage.haskell.org/trac/hackage/ticket/421), reported by @dcoutts on 2008-12-01)

It is possible to build ghc without support for object splitting. In such a case Cabal should refuse to work if the user enables split objs.

All one has to do is to parse the output of <tt>ghc --info</tt> (which is in a <tt>Read/Show</tt> format).

See also #24.

**Comment by BardurArantsson** [comment]
Last report 2008. Is split-obj a thing anymore and is this remotely relevant to current Cabal?

**Comment by 23Skidoo** [comment]
Yes, `split-objs` is still a thing; the issue should be actually quite easy to fix.
